nrespipe/trace_orders.py

```diff
@@ -0,0 +1,1090 @@
+
+    Parameters
+    ----------
+    intensities : 1D array
+    smooth : (opt) int
+        Median-smoothing is performed over the column slice, with a kernel size of `smooth`. If 0, no smoothing.
+        If not 0, `smooth` must be an odd number. Occurs before background removal, if applicable.
+    remove_background : (opt) bool
+        If true, removes the background
+
+    Returns
+    -------
+    flux: 1D array
+        Summed intensity values within the column slice.
+    """
+
+    if smooth > 0:
+        intensities = medfilt(intensities, kernel_size=smooth)
+
+    if remove_background:
+        intensities = get_remove_background(intensities)
+
+    flux = sum(intensities)
+    return flux
+
+
+def get_column(image, column,
+               center=None, halfwidth=None,
+               lower_limit=None, upper_limit=None,
+               remove_background=True, smooth=0):
+    """
+    Retrieve a columnar slice from the image.
+
+    Parameters
+    ----------
+    image : 2D array
+    column : int
+        Index of the column that will contain the slice.
+    center : (opt) int or float = None
+        Index of the central row of the slice. Floats are rounded to the nearest int. Must be provided in conjunction
+        with `halfwidth`.
+    halfwidth : (opt) int = None
+        The halfwidth of the slice. Fullwidth will be 1 + (halfwidth * 2). Must be provided in
+        conjunction with `center`.
+    lower_limit : (opt) int or float
+        The lower row index of the slice (inclusive). Floats are rounded to the nearest int. Must be provided in
+        conjunction with `upper_limit`.
+    upper_limit : (opt) int or float
+        The upper row index of the slice (exclusive). Floats are rounded to the nearest int. Must be provided in
+        conjunction with `lower_limit`.
+    smooth : (opt) int
+        Median-smoothing is performed over the slice, with a kernel size of `smooth`. If 0, no smoothing. If not 0,
+        `smooth` must be an odd number. Occurs before background removal, if applicable.
+    remove_background : (opt) bool
+
+    Returns
+    -------
+    indices : 1D array
+        Indices for the slice within the original full-frame image.
+    intensities : 1D array
+        Intensity values within the slice.
+    """
+
+    if smooth > 0 and smooth % 2 == 0:
+        smooth += 1
+        warnings.warn('`smooth` should be an odd number. Increased value to {}.'.format(smooth))
+
+    if center is not None and halfwidth is not None:
+        center = int(np.round(center))
+        indices = np.arange(center - halfwidth, center + halfwidth + 1)
+    elif lower_limit is not None and upper_limit is not None:
+        lower_limit = int(np.round(lower_limit))
+        upper_limit = int(np.round(upper_limit))
+        indices = np.arange(lower_limit, upper_limit)
+    else:
+        indices = None
+
+    if indices is not None:
+        intensities = image[indices[0]:indices[-1] + 1, column]
```

Comment:
This does not seem to work for fpacked images. It appears that image is none for that case.

Reply:
I'm not sure what fpacked images are. I didn't test for this case. Can you provide an example that I can play with?

Reply:
fpacked images are compressed images. You can find one from any of the flat field images available through archive.lco.global.

Reply:
the .fits.fz files? Yes, I see this is a problem. I'll find a fix.

Reply:
Should be fixed in latest commit.